test/unit/spec/kbaseNarrativeSpec.js

```diff
@@ -1,20 +1,163 @@
 /*global define*/
 /*global describe, it, expect*/
 /*global jasmine*/
-/*global beforeEach, afterEach*/
+/*global beforeEach, afterEach, beforeAll*/
 /*jslint white: true*/
 
-define (
-    [
-        'narrativeConfig',
-        'kbaseNarrative'
-    ], function(
-        Config,
-        Narrative
-    ) {
-    describe('Test the kbaseNarrative module', function() {
-        it('Should do things', function() {
+define ([
+    'jquery',
+    'narrativeConfig',
+    'kbaseNarrative',
+    'base/js/namespace',
+    'narrativeLogin'
+], (
+    $,
+    Config,
+    Narrative,
+    Jupyter,
+    NarrativeLogin
+) => {
+    'use strict';
 
+    const DEFAULT_FULLY_LOADED = false,
+        DEFAULT_WRITABLE = false,
+        DEFAULT_NOTEBOOK_NAME = 'some notebook',
+        FAKE_TOKEN = 'foo';
+
+    describe('Test the kbaseNarrative module', () => {
+        let loginDiv = $('<div>');
+
+        beforeEach(async () => {
```

Comment:
Testing was challenging.

beforeEach mocks up the /token and /me calls to auth - these define the auth session, which gets consumed by Narrative.init. So they're kinda necessary. The mock returns a truncated object with enough info to get through.

It also mocks a "Jupyter" object. We aren't testing the Jupyter machinery, but need some of it - specifically the kernel - to be present. Each of those elements are either called by the Narrative.init object, or the jobCommChannel. Mocking the comm channel is probably a better solution, but due to the nature of AMD load orders and testing, that's not trivial. This is verbose, but much easier to control with few side effects.